curp/src/server/raw_curp/mod.rs

```diff
@@ -519,6 +530,51 @@ impl<C: 'static + Command, RC: RoleChange + 'static> RawCurp<C, RC> {
         Ok((st_w.term, self_spec_pool))
     }
 
+    /// Handle `pre_vote`
+    /// Return `Ok(term, spec_pool)` if the vote is granted
+    /// Return `Err(term)` if the vote is rejected
+    pub(super) fn handle_pre_vote(
```

Comment:
Could we merge this method into `handle_vote()`? They are very similar.

Reply:
I don't think so. They are somewhat similar, but there are more differences, and it is easier to understand in two methods